wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -69,6 +71,39 @@ protected final Object writeReplace() throws ObjectStreamException {
     return new MessageSerializedForm<>(this, (Class<Message>) getClass());
   }
 
+  /** Returns a new builder for this class, initialized with the data in this message. */
+  public final <B extends Builder<T, B>> B newBuilder() {
```

Comment:
should this be public? Useful for metaprogramming. Regular programming can just use the existing Builder constructor.